app/models/concerns/host_common.rb

```diff
@@ -201,16 +201,25 @@ def classes_in_groups
     end
   end
 
-  def individual_puppetclasses
+  # Returns PuppetClasses of a Host or Hostgroup
+  #
+  # Params:
+  #  * +with_environments+: {Boolean} if possible environments should be included
+  #  * +in_environments+: {Boolean} to consider only classes in environments
+  #
+  def individual_puppetclasses(with_environments: false, in_environment: true)
```

Comment:
I'm not sure I understand why this method has arguments if it's only called once (in the view), where 'with_environments' is true and 'in_environment' is false. I'd suggest to either:

* remove the arguments completely (adds complexity to the method that I don't think we're using anywhere)
* make the arguments default to what we're using